chain/vm/vm.go

```diff
@@ -335,6 +347,14 @@ func (vm *VM) ApplyMessage(ctx context.Context, msg *types.Message) (*ApplyRet,
 
 	ret, actorErr, rt := vm.send(ctx, msg, nil, msgGasCost)
 
+	{
+		actorErr2 := rt.chargeGasSafe(rt.Pricelist().OnChainReturnValue(len(ret)))
+		if actorErr == nil {
+			//TODO: Ambigous what to do in this case
```

Comment:
I think we fail the execution, revert, and have no return value. So what youre doing here seems correct